pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -820,6 +823,27 @@ func (og *operationGenerator) GenerateVerifyControllerAttachedVolumeFunc(
 	}, nil
 }
 
+func (og *operationGenerator) GenerateVerifyControllerDetachedVolumeFunc(
+	volumeToDetach AttachedVolume,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) (func() error, error) {
+	return func() error {
+		if !volumeToDetach.PluginIsAttachable {
+			// If the volume does not implement the attacher interface, it is
+			// assumed to be detached and the actual state of the world is
+			// updated accordingly.
+			actualStateOfWorld.MarkVolumeAsDetached(volumeToDetach.VolumeName, volumeToDetach.NodeName)
+			return nil
+		}
+
+		glog.Infof("Controller detached volume %q (spec.Name: %q) devicePath: %q",
+			volumeToDetach.VolumeName,
+			volumeToDetach.VolumeSpec.Name(),
+			volumeToDetach.DevicePath)
+		actualStateOfWorld.MarkVolumeAsDetached(volumeToDetach.VolumeName, volumeToDetach.NodeName)
+		return nil
```

Comment:
It's difficult to get the globally mounted state inside of operation_executor, since it's  private state of the kubelet's actualStateOfTheWorld. We need to requery the asw inside of the callback to check if globally mounted changed (can't just pass it in from reconciler, since it's a copy). I can either open up the volume's asw interface, or can check if the device is mounted directly. I'm inclined to do the latter, since it's less dependent on state management in kubelet, and probably a more reliable check of the real state of things.